mistralrs-core/src/sequence.rs

```diff
@@ -231,10 +234,30 @@ impl Sequence {
         self.state.set(state);
     }
 
-    pub fn is_done(&self, tok: u32, eos_tok: u32, max_model_len: usize) -> Option<StopReason> {
+    pub fn generated_text(&self, tokenizer: &Tokenizer) -> String {
+        let token_ids = self
+            .logprobs()
+            .iter()
+            .map(|lp| lp.token)
+            .collect::<Vec<_>>();
+        let text = tokenizer
+            .decode(&token_ids, false)
+            .expect("Failed to decode tokens generated by model");
```

Comment:
Can the tokenizer fail here? Like if the last token doesn't complete a valid UTF codepoint yet